streaming/src/main/scala/org/apache/spark/streaming/ui/BatchPage.scala

```diff
@@ -288,7 +290,14 @@ private[ui] class BatchPage(parent: StreamingTab) extends WebUIPage("batch") {
     val batchTime = Option(request.getParameter("id")).map(id => Time(id.toLong)).getOrElse {
       throw new IllegalArgumentException(s"Missing id parameter")
     }
-    val formattedBatchTime = SparkUIUtils.formatDate(batchTime.milliseconds)
+    val batchTimeFormat =
+      if (streamingListener.batchDuration < 1000) {
+        new SimpleDateFormat("yyyy/MM/dd HH:mm:ss.SSS")
+      } else {
+        // If batchInterval >= 1 second, don't show milliseconds
+        new SimpleDateFormat("yyyy/MM/dd HH:mm:ss")
+      }
```

Comment:
these few lines are duplicated in a few places. Can you put it in `streaming.ui.UIUtils`? I will also find this useful in my code.